src/Bridges/bridge_optimizer.jl

```diff
@@ -1160,7 +1160,11 @@ function _modify_bridged_function(
     change::MOI.AbstractFunctionModification,
 )
     if is_bridged(b, ci_or_obj)
-        MOI.modify(recursive_model(b), bridge(b, ci_or_obj), change)
+        try
+            MOI.modify(recursive_model(b), bridge(b, ci_or_obj), change)
+        catch
+            MOI.throw_modify_not_allowed(ci_or_obj, change)
+        end
```

Comment:
@blegat do you have any ideas for this? We can't add a generic fallback for the `::Bridge` case of the second argument because then we don't know whether we're changing a constraint or an objective.

Reply:
Not sure what you mean. I would add a fallback in `Bridges/bridge.jl`. If you want to differentiate, you have `Bridges.Objective.AbstractBridge` and `Bridges.Constraint.AbstractBridge`.

Reply:
But how do we know which `ConstraintIndex` it refers to?

Reply:
Ah yes, I guess this is an annoying part of this error. This is also why you had to do a `try`-`catch` in https://github.com/jump-dev/MathOptInterface.jl/pull/2300.
I think we should allow throwing this error without knowing what is the constraint index is.
At the end of the day, after the index maps, this might not be the index of the user anyway.

Reply:
Yeah a generic `ModifyNotAllowed` would be simpler. But that could be a separate PR after this is merged?

Reply:
Yes but then open an issue or leave a comment here to keep not of that

Reply:
Opened an issue: https://github.com/jump-dev/MathOptInterface.jl/issues/2306. I'll work on it after this PR is merged